forge/csrc/passes/lower_to_mlir.cpp

```diff
@@ -187,6 +187,14 @@ class MLIRGenerator
                     }
                     return builder_.getArrayAttr(attributes);
                 }
+                else if constexpr (std::is_same_v<T, std::tuple<int, int, int>>)
```

Comment:
Does this mean that we're only handling rank 3 tuples? Can we extend it to hand N-dimensional tuples :))

Reply:
I too explored on this and couldn't find an exact way. All the models which I checked hit with this `std::tuple<int, int, int>`. I created a separate issue to explore on this  N-dimensional tuples after merging this PR. https://github.com/tenstorrent/tt-forge-fe/issues/705